src/dss/list.py

```diff
@@ -0,0 +1,54 @@
+from lightkube import Client
+from lightkube.core.exceptions import ApiError
+from lightkube.resources.apps_v1 import Deployment
+from tabulate import tabulate
+
+from dss.config import DSS_NAMESPACE, NOTEBOOK_LABEL, DeploymentState
+from dss.logger import setup_logger
+from dss.utils import get_deployment_state, get_service_url, truncate_row
+
+# Set up logger
+logger = setup_logger("logs/dss.log")
+
+
+def list_notebooks(lightkube_client: Client, wide: bool = False) -> None:
+    """
+    List the available notebooks in the DSS namespace.
+
+    Args:
+        lightkube_client (Client): The Kubernetes client.
+        wide (bool, optional): Whether to display the full information without truncation.
+                               Defaults to False.
+    """
+    try:
+        deployments = lightkube_client.list(Deployment, namespace=DSS_NAMESPACE)
+    except ApiError as e:
+        logger.error(f"Failed to list notebooks: {e}")
+        return
+
+    notebook_list = []
+    for deployment in deployments:
+        labels = deployment.metadata.labels
+        if NOTEBOOK_LABEL in labels:
+            name = deployment.metadata.name
+            image = deployment.spec.template.spec.containers[0].image
+            state = get_deployment_state(deployment, lightkube_client)
+
+            # Use state to decide what to display in the URL column
+            if state == DeploymentState.ACTIVE:
+                available_replicas = deployment.status.availableReplicas
+                url = (
+                    get_service_url(name, DSS_NAMESPACE, lightkube_client)
```

Comment:
If I have a `Notebook` deployment that does not have a `Service` and I run `dss list --kubeconfig $KUBECONFIG`, I get the following error:

```
2024-04-18 15:26:34 [ERROR] [utils] [get_service_url]: Failed to get the URL of notebook test-notebook with error code 404.
Check the debug logs for more details.
2024-04-18 15:26:34 [DEBUG] [utils] [get_service_url]: Failed to get the URL of notebook test-notebook with error: services "test-notebook" not found
2024-04-18 15:26:34 [ERROR] [main] [list_command]: An error occurred: object of type 'NoneType' has no len()
```

At minimum, `dss list` should be able to ignore this and show all other notebooks when there is one incomplete notebook present.  

Similarly, if we had a notebook `Service` without a corresponding `Deployment`, it wont show up here at all (but at least `dss list` will show all the other notebooks without issue)

Have we defined how `dss list` should behave when Notebooks are incomplete (eg: if just a notebook's deployment **or** service exist, not both)?  `dss create` has a failure message that says "Some resources might be left in the cluster. Check the status with `dss list`" which imo means `dss list` needs to show when incomplete Notebooks are deployed, too.  Or we need to change the `dss create` message

Reply:
k that sounds good as an interpretation.  Can we put that in the spec too?

Reply:
Added [section](https://docs.google.com/document/d/1v0wU8TBb-ui6RcjGRbLwA6XFZnWbuiVb2TPmEgwKejY/edit#heading=h.r2mgghl5zl7j) to spec. FOr now we can keep the messages as we want to fix how the logger messages are displayed in the stdout [here](https://github.com/canonical/data-science-stack/issues/77) is the issue.